client/my-sites/upgrades/checkout-thank-you/google-voucher-details.jsx

```diff
@@ -0,0 +1,237 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import i18n from 'lib/mixins/i18n';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import PurchaseDetails from 'components/purchase-detail';
+import PurchaseButton from 'components/purchase-detail/purchase-button';
+import TipInfo from 'components/purchase-detail/tip-info';
+import Dialog from 'components/dialog';
+import googleTermsAndConditions from './google-voucher-TC';
+import QuerySiteVouchers from 'components/data/query-site-vouchers';
+import {
+	assignSiteVoucher as assignVoucher
+} from 'state/sites/vouchers/actions';
+import { GOOGLE_AD_CREDITS } from 'state/sites/vouchers/service-types';
+import {
+	getVouchersBySite,
+	getGoogleAdCredits
+} from 'state/sites/vouchers/selectors';
+
+class GoogleVoucherDetails extends Component {
+	constructor() {
+		super();
+
+		this.state = {
+			showTermsAndConditions: false,
+			step: 0
+		};
+	}
+
+	componentWillMount() {
+		// bounds
+		this.onButtonClick = this.onButtonClick.bind( this );
+		this.onDialogCancel = this.onDialogCancel.bind( this );
+		this.onAgreeButton = this.onAgreeButton.bind( this );
+
+		const voucher = this.getVoucher();
+		if ( voucher && voucher.status === 'assigned' ) {
+			this.setState( { step: 2 } );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.googleAdCredits ) {
+			return null;
+		}
+
+		if ( this.props.googleAdCredits === nextProps.googleAdCredits ) {
+			return null;
+		}
+
+		if ( nextProps.googleAdCredits.length > 0 ) {
+			this.setState( { step: 2 } );
+		}
+	}
+
+	onButtonClick() {
+		this.incrementStep();
+	}
+
+	onDialogCancel() {
+		this.setState( {
+			showTermsAndConditions: false,
+			step: 0
+		} );
+	}
+
+	onAgreeButton() {
+		this.props.assignVoucher( this.props.selectedSite.ID, GOOGLE_AD_CREDITS );
+
+		this.setState( {
+			showTermsAndConditions: false,
+			step: 2
+		} );
+	}
+
+	incrementStep() {
+		this.setState( {
+			step: this.state.step + 1,
+			showTermsAndConditions: ( this.state.step + 1 ) === 1
+		} );
+	}
+
+	getVoucher( props = this.props ) {
+		const { googleAdCredits } = props;
+		return googleAdCredits.length > 0 ? googleAdCredits[0] : {};
+	}
+
+	renderInitialStep() {
+		return (
+			<div className="purchase-detail__body">
+				<PurchaseButton
+					onClick={ this.onButtonClick }
+					text={ i18n.translate( 'Generate Code' ) } />
```

Comment:
Instead of using i18n directly we can use `client/lib/mixins/i18m/localize` instead, which will create a new component with all the previous props plus translate.